cudamapper/src/index_gpu.cuh

```diff
@@ -747,19 +750,11 @@ void IndexGPU<SketchElementImpl>::generate_index(const io::FastaParser& parser,
 
     // *** sort sketch elements by representation ***
     // As this is a stable sort and the data was initailly grouper by read_id this means that the sketch elements within each representations are sorted by read_id
+    // TODO: consider using a CUB radix sort based function here
     thrust::stable_sort_by_key(thrust::device,
-                               representations_d.data(),
-                               representations_d.data() + representations_d.size(),
-                               rest_d.data());
-
-    // copy the data to member functions (depending on the interface desing these copies might not be needed)
-    representations_d_.resize(representations_d.size());
-    representations_d_.shrink_to_fit();
-    thrust::copy(thrust::device,
-                 representations_d.data(),
-                 representations_d.data() + representations_d.size(),
-                 representations_d_.begin());
-    representations_d.free();
+                               std::begin(representations_d_),
```

Comment:
Does it make sense to sort `sketch_elements.representations_d` + `sketch_elements.rest_d` and then move them to `representations_d_`? That way the object would remain in a valid state when an exception is thrown. It might have already destroyed `sketch_elements.representations_d` though...

Reply:
Yes, I'll implement it. But the code is a exception-safety minefield anyway...

Reply:
done

Reply:
yes, I know. :boom: So I am unsure if I should care about such things.